pkg/testmachinery/testflow/node/node.go

```diff
@@ -151,8 +153,15 @@ func (n *Node) isRootNode() bool {
 	return n.inputSource != nil
 }
 
-func (n *Node) DetermineArtifacts() []argov1.Artifact {
+func (n *Node) GetOrDetermineArtifacts() []argov1.Artifact {
+	if artifacts != nil {
+		return artifacts
+	}
 	artifactsMap := make(map[string]argov1.Artifact)
+	if n.step.ArtifactsFrom != "" {
+		artifacts = n.getPreviousNode(n.step.ArtifactsFrom).GetOrDetermineArtifacts()
```

Comment:
this is not correct, there you would get the input artifacts of the àrtifactsFrom`step but you want the output artifacts of the previous step.

Therefore I would suggest to move this functionality to this method: https://github.com/gardener/test-infra/blob/2b67714452faaa9de0a03e12192c13f3dd4e0ef4/pkg/testmachinery/testflow/flow_operations.go#L135.
Then you can just set the `artifactsFrom` artifact as we already do it with the default input artifacts.
```go
func ApplyOutputScope(steps map[string]*Step) error {
	for _, step := range steps {
		for n := range step.Nodes {
			var outputSourceNode *node.Node
			if step.ArtifactsFrom != "" {
				outputSourceNode = steps[step.ArtifactsFrom]
			} else {
				parents := n.Parents
				for len(parents) != 1 {
					parents = parents.GetParents()
					if len(parents) == 0 {
						return fmt.Errorf("no serial parent node can be found for step %s in node %s", n.Name(), step.Info.Name)
					}
				}
				outputSourceNode = parents.List()[0]
			}
			outputSourceNode.EnableOutput()
			n.SetInputSource(outputSourceNode)
		}
	}

	return nil
}
```

Reply:
okay, by chance, do you mean: `outputSourceNode = steps[step.Info.ArtifactsFrom].Nodes.List()[0]` instead of `outputSourceNode = steps[step.ArtifactsFrom]`

Reply:
ah yes your're right

Reply:
done